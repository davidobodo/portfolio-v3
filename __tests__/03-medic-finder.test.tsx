import "@testing-library/react";
import { render } from "@testing-library/react";
import ProjectsPage from "#/pages/projects/[id]";

const scrollTo = ({ top, left }: { top: number; left: number }) => {};

window.scrollTo = jest.fn().mockImplementation(scrollTo);
describe(`Medic Finder page`, () => {
	it("renders correctly", () => {
		const { asFragment } = render(<ProjectsPage id="medic-finder" title="Medic finder" />);
		expect(asFragment()).toMatchSnapshot();
	});
});
