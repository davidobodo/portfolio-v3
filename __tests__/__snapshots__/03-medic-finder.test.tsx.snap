// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Medic Finder page renders correctly 1`] = `
<DocumentFragment>
  <a
    class="link"
    data-key="nav-logo"
    href="/"
    style="outline-color: rgb(225, 223, 221); visibility: visible;"
  >
    <svg
      class="fullname logo"
      fill="none"
      height="43"
      id="logo"
      opacity="1"
      viewBox="0 0 79 43"
      width="79"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M34.2803 19H36.4746L33.043 9.13574H30.6162L27.1914 19H29.3105L30.0625 16.6074H33.5283L34.2803 19ZM31.7715 11.0225H31.8193L33.084 15.0625H30.5068L31.7715 11.0225ZM41.5947 19L44.9102 9.13574H42.6611L40.4053 16.874H40.3574L38.1016 9.13574H35.791L39.1064 19H41.5947ZM47.9863 19V9.13574H45.9219V19H47.9863ZM49.627 9.13574V19H53.3936C56.3535 19 58.0967 17.1611 58.0967 14.0303C58.0967 10.8994 56.3535 9.13574 53.3936 9.13574H49.627ZM51.6914 10.8379H53.1475C54.9658 10.8379 55.9912 11.9727 55.9912 14.0371C55.9912 16.1699 54.9932 17.291 53.1475 17.291H51.6914V10.8379Z"
        fill="#e1dfdd"
      />
      <path
        d="M2.34375 0.816406V29H13.1055C21.5625 29 26.543 23.7461 26.543 14.8008C26.543 5.85547 21.5625 0.816406 13.1055 0.816406H2.34375ZM8.24219 5.67969H12.4023C17.5977 5.67969 20.5273 8.92188 20.5273 14.8203C20.5273 20.9141 17.6758 24.1172 12.4023 24.1172H8.24219V5.67969Z"
        fill="#e1dfdd"
      />
      <path
        d="M44.1885 32C46.2939 32 47.5996 30.9131 47.5996 29.1768C47.5996 27.8711 46.6152 26.8936 45.2754 26.791V26.7363C46.3213 26.5791 47.1279 25.6768 47.1279 24.5967C47.1279 23.0859 45.9658 22.1357 44.1133 22.1357H39.8203V32H44.1885ZM41.8848 23.6738H43.5801C44.5439 23.6738 45.0977 24.125 45.0977 24.918C45.0977 25.752 44.4688 26.2373 43.3682 26.2373H41.8848V23.6738ZM41.8848 30.4619V27.5977H43.6143C44.8311 27.5977 45.4941 28.0898 45.4941 29.0127C45.4941 29.9561 44.8516 30.4619 43.6621 30.4619H41.8848ZM53.2256 21.9648C50.2998 21.9648 48.4678 23.9268 48.4678 27.0713C48.4678 30.209 50.2998 32.1709 53.2256 32.1709C56.1445 32.1709 57.9834 30.209 57.9834 27.0713C57.9834 23.9268 56.1445 21.9648 53.2256 21.9648ZM53.2256 23.6875C54.8389 23.6875 55.8711 25 55.8711 27.0713C55.8711 29.1357 54.8389 30.4414 53.2256 30.4414C51.6055 30.4414 50.5732 29.1357 50.5732 27.0713C50.5732 25 51.6123 23.6875 53.2256 23.6875ZM59.3164 22.1357V32H63.083C66.043 32 67.7861 30.1611 67.7861 27.0303C67.7861 23.8994 66.043 22.1357 63.083 22.1357H59.3164ZM61.3809 23.8379H62.8369C64.6553 23.8379 65.6807 24.9727 65.6807 27.0371C65.6807 29.1699 64.6826 30.291 62.8369 30.291H61.3809V23.8379ZM73.5693 21.9648C70.6436 21.9648 68.8115 23.9268 68.8115 27.0713C68.8115 30.209 70.6436 32.1709 73.5693 32.1709C76.4883 32.1709 78.3271 30.209 78.3271 27.0713C78.3271 23.9268 76.4883 21.9648 73.5693 21.9648ZM73.5693 23.6875C75.1826 23.6875 76.2148 25 76.2148 27.0713C76.2148 29.1357 75.1826 30.4414 73.5693 30.4414C71.9492 30.4414 70.917 29.1357 70.917 27.0713C70.917 25 71.9561 23.6875 73.5693 23.6875Z"
        fill="#e1dfdd"
      />
      <path
        d="M24.0586 13.3281C15.6992 13.3281 10.4648 18.9336 10.4648 27.918C10.4648 36.8828 15.6992 42.4883 24.0586 42.4883C32.3984 42.4883 37.6523 36.8828 37.6523 27.918C37.6523 18.9336 32.3984 13.3281 24.0586 13.3281ZM24.0586 18.25C28.668 18.25 31.6172 22 31.6172 27.918C31.6172 33.8164 28.668 37.5469 24.0586 37.5469C19.4297 37.5469 16.4805 33.8164 16.4805 27.918C16.4805 22 19.4492 18.25 24.0586 18.25Z"
        fill="#e1dfdd"
      />
    </svg>
  </a>
  <div
    class="backdrop"
  />
  <div
    class="container"
    data-key="dark-section-ref"
  >
    <div
      class="content"
    >
      <div>
        <div
          class="container"
          data-key="project-info"
        >
          <button
            aria-label="previous"
            class="desktopNavigator btnPrevious"
            value="previous"
          >
            <svg
              class="icon icon-tabler icon-tabler-chevron-left"
              fill="none"
              height="60"
              stroke="#86868b"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              width="60"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
                stroke="none"
              />
              <polyline
                points="15 6 9 12 15 18"
              />
            </svg>
          </button>
          <div
            class="content"
          >
            <section
              class="title"
              data-key="title"
            >
              <h1>
                Medic finder
              </h1>
              <button
                class="btnClose"
                data-key="close-button"
              >
                <span>
                  <strong>
                    â†™
                  </strong>
                </span>
                <span>
                  Close
                </span>
              </button>
            </section>
            <div
              class="media"
            >
              <div>
                <div
                  class="slider"
                >
                  <div
                    class="sliderInner"
                    style="transform: translateX(-0px);"
                  >
                    <div
                      class="item"
                      style="background-image: url(/images/projects/medic-finder/home.jpg);"
                    />
                    <div
                      class="item"
                      style="background-image: url(/images/projects/medic-finder/results.jpg);"
                    />
                    <div
                      class="item"
                      style="background-image: url(/images/projects/medic-finder/history.jpg);"
                    />
                  </div>
                  <button
                    class="control next"
                    data-action=">"
                  >
                    <svg
                      class="icon icon-tabler icon-tabler-chevron-right"
                      fill="none"
                      height="60"
                      stroke="#86868b"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="1.5"
                      viewBox="0 0 24 24"
                      width="60"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                        stroke="none"
                      />
                      <polyline
                        points="9 6 15 12 9 18"
                      />
                    </svg>
                  </button>
                </div>
                <div
                  class="indicators"
                >
                  <button
                    class="indicator active"
                    data-slide="0"
                  />
                  <button
                    class="indicator"
                    data-slide="1"
                  />
                  <button
                    class="indicator"
                    data-slide="2"
                  />
                </div>
              </div>
            </div>
            <section
              class="about"
              data-key="about"
            >
              <h2>
                About this project
              </h2>
              <div
                class="details"
              >
                
		
                <p>
                  Easily find Hospitals, Clinics, Pharmacies and Health care centers around any place. This project was built as a requirement to get into the 
                  <a
                    href="https://www.enye.tech/"
                    target="_blank"
                  >
                    enye
                  </a>
                   internship programme, 
		however it's importance is far beyond just a requirement to get into the internship.
		
                </p>
                
		
                <p>
                  This project has 
                  <span>
                    two iterations
                  </span>
                  , initial iteration was built with a full authentication process as well as a backend infrastructure to store user's credentials and users search history. 
		However, for easy accessibility to everyone, these have been removed in the current iteration.
		 
                </p>
                
		
              </div>
              <div
                class="roles"
              >
                <h3>
                  Role in Project
                </h3>
                <ul>
                  <li>
                    UI/UX Designer
                  </li>
                  <li>
                    Front End Developer
                  </li>
                  <li>
                    Back End Developer
                  </li>
                </ul>
              </div>
              <div
                class="responsibilities"
              >
                <h3>
                  My Responsibilities & Features I Implemented
                </h3>
                <div>
                  
		
                  <ul>
                    
			
                    <li>
                      Designed the entire application using figma.
                    </li>
                    
			
                    <li>
                      Converted all mockup designs into 
                      <span>
                        pixel perfect pages
                      </span>
                       for all screen sizes.
                    </li>
                    
			
                    <li>
                      Implemented an authentication process flow using Firebase (now deprecated in current iteration).
                    </li>
                    
			
                    <li>
                      Made use of Google's 
                      <a
                        href="https://firebase.google.com"
                        target="_blank"
                      >
                        Cloud Firestore NoSQL database
                      </a>
                       to store user's previous searches (now refactored in current iteration to use local storage).
                    </li>
                    
			
                    <li>
                      Integrated 
                      <a
                        href="https://developers.google.com/maps"
                        target="_blank"
                      >
                        Google Maps Platform
                      </a>
                       to help give both autocomplete functionality when looking for places and display the map with the needed results.
                    </li>
                    
			
                    <li>
                      Connect to database using GraphQL.
                    </li>
                    
		
                  </ul>
                  
		
                </div>
              </div>
            </section>
            <section
              class="tech"
              data-key="tech"
            >
              <h2>
                Technical Sheet
              </h2>
              <p>
                Some noteworthy technologies I got involved with while working on this project:
              </p>
              <ul>
                <li>
                  Figma
                </li>
                <li>
                  HTML
                </li>
                <li>
                  CSS
                </li>
                <li>
                  SASS/SCSS
                </li>
                <li>
                  React.js
                </li>
                <li>
                  Vanilla Javascript
                </li>
                <li>
                  Typescript
                </li>
                <li>
                  Redux
                </li>
                <li>
                  Material UI
                </li>
                <li>
                  Firebase
                </li>
                <li>
                  GraphQL
                </li>
                <li>
                  Node.js
                </li>
                <li>
                  Express.js
                </li>
                <li>
                  Google Maps
                </li>
                <li>
                  Netlify
                </li>
              </ul>
            </section>
            <div
              class="links"
              data-key="buttons"
            >
              <a
                href="https://medic-finder.netlify.app/"
                target="_blank"
              >
                Visit site
                <svg
                  class="icon icon-tabler icon-tabler-external-link"
                  fill="none"
                  height="24"
                  stroke="#86868b"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                    stroke="none"
                  />
                  <path
                    d="M11 7h-5a2 2 0 0 0 -2 2v9a2 2 0 0 0 2 2h9a2 2 0 0 0 2 -2v-5"
                  />
                  <line
                    x1="10"
                    x2="20"
                    y1="14"
                    y2="4"
                  />
                  <polyline
                    points="15 4 20 4 20 9"
                  />
                </svg>
                 
              </a>
              <a
                href="https://github.com/obododavid/Medic-Finder"
                target="_blank"
              >
                Github repo 
                <svg
                  class="icon icon-tabler icon-tabler-brand-github"
                  fill="none"
                  height="24"
                  stroke="#86868b"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                    stroke="none"
                  />
                  <path
                    d="M9 19c-4.3 1.4 -4.3 -2.5 -6 -3m12 5v-3.5c0 -1 .1 -1.4 -.5 -2c2.8 -.3 5.5 -1.4 5.5 -6a4.6 4.6 0 0 0 -1.3 -3.2a4.2 4.2 0 0 0 -.1 -3.2s-1.1 -.3 -3.5 1.3a12.3 12.3 0 0 0 -6.2 0c-2.4 -1.6 -3.5 -1.3 -3.5 -1.3a4.2 4.2 0 0 0 -.1 3.2a4.6 4.6 0 0 0 -1.3 3.2c0 4.6 2.7 5.7 5.5 6c-.6 .6 -.6 1.2 -.5 2v3.5"
                  />
                </svg>
                 
              </a>
            </div>
          </div>
          <button
            aria-label="previous"
            class="desktopNavigator btnNext"
            value="next"
          >
            <svg
              class="icon icon-tabler icon-tabler-chevron-right"
              fill="none"
              height="60"
              stroke="#86868b"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              width="60"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
                stroke="none"
              />
              <polyline
                points="9 6 15 12 9 18"
              />
            </svg>
          </button>
          <div
            class="mobileNavigator"
          >
            <button
              aria-label="previous"
              value="previous"
            >
              <svg
                class="icon icon-tabler icon-tabler-chevron-left"
                fill="none"
                height="60"
                stroke="#86868b"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                width="60"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                  stroke="none"
                />
                <polyline
                  points="15 6 9 12 15 18"
                />
              </svg>
            </button>
            <button
              aria-label="previous"
              value="next"
            >
              <svg
                class="icon icon-tabler icon-tabler-chevron-right"
                fill="none"
                height="60"
                stroke="#86868b"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                width="60"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                  stroke="none"
                />
                <polyline
                  points="9 6 15 12 9 18"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
      <div
        class="noise"
      />
    </div>
    <div
      class="container undefined"
      data-key="radial-gradient"
    />
  </div>
  <div
    class="noise"
  />
  <footer
    class="wrapper"
  >
    <div
      class="container"
      data-key="contact-form"
      style="position: fixed;"
    >
      <div
        class="containerInner"
      >
        <div
          class="leftSection"
        >
          <div
            class="top"
          />
          <div>
            <div
              class="helpfulLinks"
            >
              <h3>
                Quick Links
              </h3>
              <ul>
                <li>
                  <a
                    href="/"
                  >
                    <span>
                      Home
                    </span>
                  </a>
                </li>
                <li
                  class="line"
                />
                <li>
                  <a
                    href="/projects"
                  >
                    <span>
                      Projects
                    </span>
                  </a>
                </li>
                <li
                  class="line"
                />
                <li>
                  <a
                    href="/letters"
                  >
                    <span>
                      Letters 
                    </span>
                  </a>
                </li>
              </ul>
            </div>
            <div
              class="helpfulLinks"
            >
              <h3>
                Extras
              </h3>
              <ul>
                <li>
                  <a
                    data-link="resume"
                    href="https://drive.google.com/file/d/1dVxGS3654jFz_YiWrkrCDU93ISZSj_lc/view?usp=sharing"
                    target="_blank"
                  >
                    <span>
                      Resume
                    </span>
                  </a>
                </li>
                <li
                  class="line"
                />
                <li>
                  <a
                    href="/credits"
                  >
                    <span>
                      Site credits
                    </span>
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </div>
        <div
          class="formSection"
        >
          <div
            class="title"
          >
            <h1>
              Would love to hear from 
              <br />
               you â†“.
            </h1>
            <p>
              Iâ€™m currently interested in aÂ 
              <span>
                Full-time Front-end developer role (remote)
              </span>
               with a major on Â 
              <span>
                React.js Framework
              </span>
              , but still open to other opportunities. However, if you have other requests or questions, donâ€™t hesitate to use the form.
            </p>
          </div>
          <form
            class="form"
            novalidate=""
          >
            <div
              class="twoColumns"
            >
              <div
                class="formField"
              >
                <label
                  for="name"
                >
                  Name
                </label>
                <input
                  id="name"
                  name="name"
                  placeholder="Name"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="formField"
              >
                <label
                  for="email"
                >
                  Email
                </label>
                <input
                  id="email"
                  name="email"
                  placeholder="Email"
                  type="email"
                  value=""
                />
              </div>
            </div>
            <div
              class="formField"
            >
              <label
                for="subject"
              >
                Subject
              </label>
              <input
                id="subject"
                name="subject"
                placeholder="Subject"
                type="text"
                value=""
              />
            </div>
            <div
              class="formField"
            >
              <label
                for="message"
              >
                Message
              </label>
              <textarea
                cols="30"
                id="message"
                name="message"
                placeholder="Message"
                rows="10"
              />
            </div>
            <div
              class="btnWrapper"
            >
              <button
                aria-label="Submit"
                disabled=""
                type="submit"
              >
                <span>
                  Submit
                </span>
                <svg
                  fill="none"
                  viewBox="0 0 60 60"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="m.819 50.513 8.307 8.238 38.423-38.454-.059 28.89h11.638V.424H10.47l-.14 11.564h28.983L.819 50.513Zm55.31-47.09v42.764V3.424Z"
                    fill="currentColor"
                    fill-rule="evenodd"
                  />
                </svg>
              </button>
            </div>
          </form>
        </div>
      </div>
      <div
        class="social"
      >
        <ul>
          <li>
            <a
              data-link="linkedin"
              href="https://www.linkedin.com/in/david-obodo-998786174/"
              target="_blank"
            >
              <span>
                Linkedin
              </span>
            </a>
          </li>
          <li>
            <a
              data-link="github"
              href="https://github.com/davidobodo"
              target="_blank"
            >
              <span>
                Github
              </span>
            </a>
          </li>
          <li>
            <a
              data-link="twitter"
              href="https://twitter.com/phitGeek"
              target="_blank"
            >
              <span>
                Twitter
              </span>
            </a>
          </li>
          <li>
            <a
              data-link="email"
              href="mailto: obododavid5@gmail.com"
              target="_blank"
            >
              <span>
                Email
              </span>
            </a>
          </li>
        </ul>
        <p>
           Â© 2022 David Obodo
        </p>
      </div>
    </div>
    <div
      class="placeholderWrapper"
    >
      <div
        style="height: 0px; display: block;"
      />
    </div>
    <div
      class="blackCurtain"
      data-key="contact-curtain"
    />
  </footer>
</DocumentFragment>
`;
